biorxiv/corpora_comparison/biorxiv_pmc_figure_generator.py

```diff
@@ -0,0 +1,486 @@
+        'xanchor': 'center',
+        'yanchor': 'top'}
+)
+fig.update_layout(uniformtext=dict(minsize=10))
+fig.show()
+
+
+# In[18]:
+
+
+fig.write_image("output/svg_files/preprint_published_comparison_treemap.svg", scale=2.5)
+
+
+# ## Line Plot
+
+# In[19]:
+
+
+full_plot_df = (
+    preprint_published_comparison
+    .assign(
+        lower_odds=lambda x: np.exp(
+            np.log(x.odds_ratio) - 1.96*(
+                # log(odds) - z_alpha/2*sqrt(1/a+1/b+1/c+1/d)
+                np.sqrt(
+                    1/x.preprint_count + 1/x.published_count + 1/x.preprint_total + 1/x.published_total
+                )
+            )
+        ),
+        upper_odds=lambda x: np.exp(
+            np.log(x.odds_ratio) + 1.96*(
+                # log(odds)+ z_alpha/2*sqrt(1/a+1/b+1/c+1/d)
+                np.sqrt(
+                    1/x.preprint_count + 1/x.published_count + 1/x.preprint_total + 1/x.published_total
+                )
+            )
+        )
+    )
+)
+full_plot_df.head()
+
+
+# In[20]:
+
+
+plot_df = (
+    full_plot_df
+    .sort_values("odds_ratio", ascending=False)
+    .head(subset)
+    .append(
+        full_plot_df
+        .sort_values("odds_ratio", ascending=False)
+        .tail(subset)
+    )
+)
+plot_df.head()
+
+
+# In[21]:
+
+
+g = (
+    p9.ggplot(plot_df, p9.aes(x="lemma", y="odds_ratio"))
+    + p9.geom_pointrange(p9.aes(ymin="lower_odds", ymax="upper_odds"), position=p9.position_dodge(width=5))
+    + p9.scale_x_discrete(limits=plot_df.sort_values("odds_ratio", ascending=True).lemma.tolist())
+    + p9.geom_hline(p9.aes(yintercept=1), linetype = '--', color='grey')
+    + p9.coord_flip()
+    + p9.theme_seaborn(context='paper')
+    + p9.theme(
+        # 1024, 768
+        figure_size=(13.653333333333334, 10.24),
+        axis_text_y=p9.element_text(family='DejaVu Sans', size=12),
+        panel_grid_minor=p9.element_blank(),
+        axis_title=p9.element_text(size=15),
+        axis_text_x=p9.element_text(size=11, weight="bold")
+    )
+    + p9.labs(x=None)
+)
+g.save("output/svg_files/preprint_published_frequency_odds.svg", dpi=75)
+print(g)
+
+
+# In[22]:
+
+
+count_plot_df = pd.DataFrame(
+    list(
+        zip(
+            plot_df.lemma.tolist(), 
+            plot_df.preprint_count.tolist(), 
+            plot_df.assign(label='preprint').label.tolist()
+
+        )
+    )
+    +
+    list(
+        zip(
+            plot_df.lemma.tolist(), 
+            plot_df.published_count.tolist(), 
+            plot_df.assign(label='published').label.tolist()
+
+        )
+    ),
+    columns=["lemma", "count", "category"]
+)
+count_plot_df.head()
+
+
+# In[23]:
+
+
+g = (
+    p9.ggplot(count_plot_df, p9.aes(x="lemma", y="count"))
+    + p9.geom_col(position=p9.position_dodge(width=0.5))
+    + p9.coord_flip()
+    + p9.facet_wrap("category")
+    + p9.scale_x_discrete(limits=plot_df.sort_values("odds_ratio", ascending=True).lemma.tolist())
+    + p9.scale_y_continuous(labels=custom_format('{:,g}'))
```

Comment:
It makes sense to sort by odds ratio here, but the order of the image will be confusing unless it ends up next to the line plots in the publication

Reply:
>It makes sense to sort by odds ratio here, but the order of the image will be confusing unless it ends up next to the line plots in the publication

Actually the line plots and bar charts are suppose to be aligned together in one figure. I don't know what is going on with my notebook versioning, but I'll update the notebook to clear confusion.